### DISCUSSION TITLE
librewolf homebrew installation is damaged

### DISCUSSION THREAD

**Discussion by BitSynKed** [Homebrew/discussions#3348] [Casks]
I think this occurred from the latest update to librewolf. Whenever I try to open it, the computer tells me its damaged and should be trashed.
I tried reinstalling it, and when that didn't work I also forced the cache to remove all installs so that the .dmg file would be downloaded again. Reinstalling with that newly downloaded librewolf is still damaged.
I then downloaded it manually from librewolf's website and installed to Applications, keeping both versions. The manual download works without an issue, the homebrew copy is still corrupted.
Is anyone having similar issues? Should I submit an issue to the homebrew-cask branch?

**Comment by flowoy96** [comment]
Hey, this is due to macOS's Gatekeeper. Just reinstall LibreWolf with the following flag:

`brew reinstall librewolf --no-quarantine`

This should solve the problem :)

**Reply by SMillerDev** [reply]
Better option would be to have librewolf sign their app correctly though.

**Reply by Bug-Reaper** [reply]
2023 and it's still like this. It's a shame we can't get a properly signed libre-wolf dmg as unsandboxed web browser that is slow to recieve security updates feels very meh. That said, huge shoutouts to the libre-wolf team. This is a labor of love and apple does make proper sign a PITA especially when multiple binaries are bundled. Perhaps one day I'll look to properly sign/notarize this.

**Reply by Jolg42** [reply]
Same issue here with latest version `117.0-1/librewolf-117.0-1-macos-arm64-package.dmg`

I tried `brew install --cask librewolf` but got this damaged issue.

`brew reinstall librewolf --no-quarantine` was necessary to make it successful.

**Reply by Jolg42** [reply]
It looks like the tracking issue was closed

https://gitlab.com/librewolf-community/browser/macos/-/issues/19
Now moved to https://codeberg.org/librewolf/issues/issues/710

**Reply by AllThePi** [reply]
Just commenting to say this is still the case for me on 1/24/24

**Reply by justinnichols** [reply]
And still happening on Sonoma 14.4 on an M2 Macbook Air.  If, perchance, any Librewolf devs see this...if we're supposed to install with `--no-quarantine` then perhaps the installation instructions for macOS on the main website should state this.

**Reply by gromgit** [reply]
> the installation instructions for macOS on the main website should state this.

It does now: https://librewolf.net/installation/macos/#homebrew, which in turn points to https://librewolf.net/docs/faq/#why-is-librewolf-marked-as-broken

**Reply by k4u5hik** [reply]
Thank you this worked on Tahoe 26.2 (25C56).

**Comment by lightnin** [comment]
Still happening Sonomo 14.5 on M3. 
Would it be possible to change the brew instruction set so that we did not have to reinstall with  the --no-quarantine flag after each update? Or does it require action by librewolf maintainers?

**Reply by purajit** [reply]
There isn't much that can be done, there's no way to tell Brew to install it with the same flags again, and it's not part of the formula configuration.

**Comment by mrtysn** [comment]
Same issue on M2 Max chip with MacOS Sonoma 14.6.1. The fix below still works.

`brew reinstall librewolf --no-quarantine
`

**Comment by jchannon** [comment]
Just had to use the fix today 😢

**Comment by d33p-dev** [comment]


**Reply by SMillerDev** [reply]
It's only fixable by librewolf

**Reply by d33p-dev** [reply]
Yep sorry.  I was speed running through different things and didn't even realize I stumbled onto the HomeBrew GH and not LibreWolf.  Sorry, I'll remove comment.

**Comment by SniffleSneeze** [comment]
I can confirm this still work to bypass macOS security on 15.3.1 sequoia: 
```
brew reinstall librewolf --no-quarantine 
```

**Reply by SMillerDev** [reply]
It doesn't actually fix anything, it bypassed macOS security to install the cask without enabling the usual security triggering.

**Reply by SniffleSneeze** [reply]
yeah apologies, I mainly wanted to bump up the tread, knowing that in the coming days LibreWolf and Brave are gonna probably be looked into.

**Comment by moonpost** [comment]
confirming latest fresh install as of today and it still immediately comes up with "librewolf is damaged..."

**Comment by purajit** [comment]
FYI another issue is that the uninstall + reinstall leaves some old metadata around permissions, making it seemingly impossible to allow screensharing. It's fixed by deleting the old permission entry and re-requesting; I now have this in my shell rc and run it after upgrades (this avoids having to reinstall or use `--no-quarantine`)
```sh
function fix_librewolf {
  xattr -d com.apple.quarantine /Applications/LibreWolf.app
  tccutil reset ScreenCapture org.mozilla.librewolf
}
```

**Comment by shoguevara** [comment]


**Reply by LouKordos** [reply]
Am i understanding correctly that there is no proper fix for this and that i need to run the reinstall command + the patches you mentioned everytime this appears?

**Reply by JordanRex** [reply]
its not a bug. its a refusal to bow down to corporate bs. its fair. not much of an effort to run a single command for free software.

**Comment by fabianschulz** [comment]
What is going to happen to the LibreWolf cask starting September 2026?

`fabian@MacBook ~ % brew upgrade librewolf`
`Warning: librewolf has been deprecated because it does not pass the macOS Gatekeeper check! It will be disabled on 2026-09-01.`

I quote from https://librewolf.net/docs/faq/ :

> (...) we do not notarize the macOS version of the browser: we don't have a paid Apple Developer license and we don't want to support this signing mechanism that is put behind a paywall without providing significant gains.

Does this mean that the LibreWolf cask will be removed from Homebrew starting September?

**Reply by SMillerDev** [reply]
It will be uninstallable from the main tap. If someone else makes a tap for it you'll still be able to use that, but the Homebrew maintainers will not maintain that

**Comment by akdev1l** [comment]
I put a script like this in `~/.local/bin/brew` - at least that way you don’t have to remember about it 


```bash
#!/bin/bash

/opt/homebrew/bin/brew "$@"
brew_exit=$?

if [ "$1" = "upgrade" ]; then
  APPLICATIONS=(
    "LibreWolf.app"
  )

  for app in "${APPLICATIONS[@]}"; do
    if [ -d "/Applications/${app}" ]; then
      printf '[%s] quarantine.fix: removing quarantine %s\n' "$(date '+%Y-%m-%d %H:%M:%S')" "$app" >&2
      xattr -dr com.apple.quarantine "/Applications/${app}"
    fi
  done
fi

exit $brew_exit
```
---

I made a tap which installs librewolf and drops a launchagent to remove the quaratine xattr: https://github.com/akdev1l/homebrew-apps

```
brew tap akdev1l/apps
brew install --cask akdev1l/apps/librewolf
```

in theory that should mean during updates we don't need to worry about it

**Comment by thekryz** [comment]
Update on this: upstream started signing the macOS builds as of release 154.0.1-2 (2026-08-26).

Release notes: https://librewolf.dev/librewolf/bsys6/releases/tag/154.0.1-2

I installed the 154.0.1-2 arm64 dmg manually and it passes the Gatekeeper check:

```
$ spctl -a -vvv /Applications/LibreWolf.app
/Applications/LibreWolf.app: accepted
source=Notarized Developer ID
origin=Developer ID Application: Calibour Limited (2S5C46ZFKS)
```

No quarantine attribute is set after installation, and the app launches without any dialog. The version the cask currently points to (154.0,2) is still `Signature=adhoc`, so the deprecation was accurate at the time — it just isn't anymore.

The cask hasn't picked up the new version, so as things stand it will still be disabled on 2026-09-01 while pointing at the last unsigned build. Might be worth someone opening a PR to bump it and drop the disable stanza before then.

**Comment by simesy** [comment]
With the new signed release I was able to reinstall to fix the issue.

```
brew uninstall librewolf
brew install --cask librewolf
```
